bundles/mapping/mapmodule/plugin/vectorlayer/VectorLayerPlugin.ol.js

```diff
@@ -590,6 +590,16 @@ Oskari.clazz.define(
                     var evt = Oskari.eventBuilder('MapLayerEvent')(layer.getId(), 'update');
                     this._sandbox.notifyAll(evt);
                 }
+                if (options.remove) {
+                    const request = Oskari.requestBuilder('RemoveMapLayerRequest')(layer.getId());
+                    this._sandbox.request(this, request);
+
+                    const mapLayerService = this._sandbox.getService('Oskari.mapframework.service.MapLayerService');
```

Comment:
We already have the service in a variable earlier in this function. Make that const instead and drop this line. Also I would move the remove portion to the start of the function and return early if removal was requested to make it easier to read. Also that way any layer properties are not unnecessarily updated for any reason just to get the removal done.

Reply:
Check!